util/common.pri

```diff
@@ -11,39 +11,30 @@ PENCIL2D_RELEASE {
     DEFINES += PENCIL2D_RELEASE_BUILD
 }
 
-CONFIG += c++11
-
-win32-g++ {
-    QMAKE_CXXFLAGS += -std=c++11
+CONFIG += strict_c strict_c++
+greaterThan(QT_MAJOR_VERSION, 5) {
+    CONFIG += c++17
+} else {
+    CONFIG += c++11
 }
 
-win32-msvc* {
-    QMAKE_CXXFLAGS += /MP /utf-8
-    CONFIG(release,debug|release) {
-        QMAKE_CXXFLAGS += /Gy /GL
-        CONFIG += ltcg
-        CONFIG += force_debug_info
-    }
+# utf8_source is only for Qt 5, it is the default since Qt 6
+CONFIG += msvc_mp utf8_source
+
+win32-msvc*:CONFIG(release,debug|release) {
+    QMAKE_CXXFLAGS += /Gy /GL
+    CONFIG += ltcg
```

Comment:
As you are working on this part, it would be nice to remove `/Gy /GL` and `CONFIG+=ltcg` since they don't make a noticable difference if Qt itself is not compiled with LTCG enabled,